components/brave_shields/browser/tracking_protection_service_browsertest.cc

```diff
@@ -190,20 +199,23 @@ IN_PROC_BROWSER_TEST_F(TrackingProtectionServiceTest, StorageTrackingBlocked) {
 }
 
 IN_PROC_BROWSER_TEST_F(TrackingProtectionServiceTest, StorageTrackingAllowed) {
-  const base::CommandLine& command_line =
-      *base::CommandLine::ForCurrentProcess();
   if (!TrackingProtectionHelper::IsSmartTrackingProtectionEnabled()) {
     return;
   }
   ASSERT_TRUE(InstallTrackingProtectionExtension());
 
-  GURL url = embedded_test_server()->GetURL("example.com",
+  // example.com is not in the StorageTrackingProtection list
+  GURL redirect_url = embedded_test_server()->GetURL("example.com",
     kStoragePage);
-  NavigateParams params(browser(), url, ui::PAGE_TRANSITION_IS_REDIRECT_MASK);
-  params.disposition = WindowOpenDisposition::CURRENT_TAB;
-  ui_test_utils::NavigateToURL(&params);
+
+  GURL url = embedded_test_server()->GetURL(std::string("/server-redirect?") +
+    redirect_url.spec());
+
+  ui_test_utils::NavigateToURL(browser(), url);
   content::WebContents* contents = browser()->tab_strip_model()->
     GetActiveWebContents();
+  ASSERT_TRUE(content::WaitForLoadStop(contents));
```

Comment:
`ui_test_utils::NavigateToURL` is blocking, so you do not need to do `content::WaitForLoadStop`